java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -2184,7 +2226,7 @@ public final ColumnVector build() {
         throw new IllegalStateException("Cannot reuse a builder.");
       }
       ColumnVector cv = new ColumnVector(type, tsTimeUnit,
-          currentIndex, nullCount, data, valid);
+          currentIndex, nullCount, data, valid, offsets);
       try {
         cv.ensureOnDevice();
         built = true;
```

Comment:
At build time, we may detect that the host buffer is bigger than required. Does it make sense to `realloc` at that point?

Reply:
In the future we may want to, but realloc is not really supported right now in an efficient way.

Reply:
Sounds good.